earth_enterprise/src/fusion/tools/boxfilter.cpp

```diff
@@ -104,39 +104,82 @@ const uchar *BoxFilterTiledImage::GetImageTile(int tile_x, int tile_y,
   }
 }
 
+// Computes the left SAT element of a given row and advances the pointer. We
+// don't have a similar function for the top row of the SAT because the
+// calculation will just wrap around the horizontal band which is set to 0,
+// so the extra elements will have no effect.
+static inline
+void ComputeLeftSAT(int **elem,  // pointer to element in SAT table
+                    uchar value,  // corresponding image value
+                    int prev_row_offset) {  // offset to previous row elem
+  **elem =
+      + *(*elem + prev_row_offset)
+      + value;
+  ++(*elem);
+}
 
 // Computes a single SAT element given current values and an offset to
 // the previous row of values in the table.
-inline void ComputeSAT(int *elem,  // pointer to element in SAT table
-                       uchar value,  // corresponding image value
-                       int prev_row_offset) {  // offset to previous row elem
-  *elem = - *(elem + prev_row_offset - 1) + *(elem + prev_row_offset)
-          + *(elem - 1)                   + value;
+static inline
+void ComputeSAT(int *elem,  // pointer to element in SAT table
```

Comment:
can we unit test this function/method 'ComputeSAT' with positive cases and negative ones to simulate going outside the matrix bounds ?

Reply:
I could write some positive test cases, but this function doesn't handle going outside the matrix bounds, so I'm not sure what the point of negative test cases would be.

Reply:
My thinking here is that these functions are not part of the interface of this module.  Since they are declared inline, they cannot be called from other files.  Since they're not part of the interface, I decided they are covered sufficiently by the existing unit tests.  In any case, it's impossible to unit test these functions without removing the inline modifier or refactoring some other way, which I think is not justified in this case.